### PR TITLE
Implement pause/resume partitions

### LINKED ISSUE #55: Ability to pause partitions
When there is an business logic error (e.g. due to unexpected message content) I would like to make progress at least on other partitions. Therefore I propose the following pseudo-code as a draft, implementing the `pause` method on `Rdkafka::Consumer`:

```ruby
loop do
  message = consumer.poll
  process(message) if message
rescue Rdkafka::RdkafkaError => e
  raise unless e.is_partition_eof?
rescue => e
  consumer.pause(message.topic, message.partition)
end
```

For reference:
 * [Kafka consumer docs on `pause`](https://kafka.apache.org/10/javadoc/org/apache/kafka/clients/consumer/KafkaConsumer.html#pause-java.util.Collection-)
 * [corresponding `librdkafka` binding](https://docs.confluent.io/3.2.1/clients/librdkafka/rdkafka_8h.html#a293cb2aac290c157702d3b82f5c14fce)

### PR DESCRIPTION
**Pull request by dmexe**
Add bindings to:

- rd_kafka_pause_partitions
- rd_kafka_resume_partitions

Fixes #55

### CODE REVIEW

File: lib/rdkafka/consumer.rb [hunk 1]
```diff
@@ -60,6 +60,42 @@ def unsubscribe
...
+    #
+    # @return [nil]
+    def resume(list)
+      unless list.is_a?(TopicPartitionList)
+        raise TypeError.new("list has to be a TopicPartitionList")
+      end
+      tpl = list.to_native_tpl
+      response = Rdkafka::Bindings.rd_kafka_resume_partitions(@native_kafka, tpl)
+      if response != 0
+        raise Rdkafka::RdkafkaError.new(response, "Error resume '#{list.to_h}'")
```
Comment (thijsc) [inline lib/rdkafka/consumer.rb]: I'd like to add an optional list field to the error class, so you can put the actual object in there for further inspection.

Comment (thijsc) [reply lib/rdkafka/consumer.rb]: Or maybe subclass error to `RdkafkaErrorWithTpl` or something like that.

Comment (dmexe) [reply lib/rdkafka/consumer.rb]: I've carefully looked at `TopicPartitionList` and I think  the first we need is a flattened  representation of `TopicPartitionList` which would be the array of TopicPartition structures

```
class TopicPartition
  attr :topic
  attr :partition
  optional :offset
  optional :err_t
end
```

So, per a partition error could be represented as `Hash<TopicPartition, RdkafkaError>`, which could be used inside `RdkafkaErrorWithTpl `

Comment (thijsc) [reply lib/rdkafka/consumer.rb]: That would indeed be nice, I'm not a huge fan of how rdkafka combines so many pieces of information in a single struct. Would make sense to create a dedicated object.

**Pull request merged**